NoteFind/View Controllers/ProfileViewController.m

```diff
@@ -31,10 +47,70 @@ - (void)setUpUserInfo {
     self.username.text = [NSString stringWithFormat:@"@%@",currentUser[@"username"]];
     self.userSchool.text = currentUser[@"school"];
     self.userMajor.text = currentUser[@"major"];
+        
+}
+
+- (void)getOnlineNotes {
+    PFQuery *noteQuery = [Note query];
+    [noteQuery orderByDescending:@"createdAt"];
+    [noteQuery includeKey:@"author"];
     
+    [noteQuery findObjectsInBackgroundWithBlock:^(NSArray<Note *> *notes, NSError *error) {
+        if (notes) {
+            self.onlineNotes = (NSMutableArray *)notes;
+            [self.collectionView reloadData];
+        } else {
+            [ErrorAlerts showAlertWithTitle:@"couldn't load notes" withMessage:@"failure loading notes. please refresh and try again." withVC:self];
+        }
+    }];
+}
+
+- (void)getOfflineNotes {
+    OfflineDataManager *dataManager = [[OfflineDataManager alloc]init];
+    self.offlineNotes = (NSMutableArray *)[dataManager getOfflineNotes];
+    [self.collectionView reloadData];
+}
+
+
+- (UICollectionViewCell *)collectionView:(nonnull UICollectionView *)collectionView cellForItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    if (self.filterSegmentCtrl.selectedSegmentIndex == 0) {
+        OnlineGridCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"OnlineGridCell" forIndexPath:indexPath];
+        cell.gridNote = self.onlineNotes[indexPath.row];
+        cell.notePost.file = cell.gridNote.note;
+        [cell.notePost loadInBackground];
```

Comment:
Is this reloading the post object from parse? This will execute every time the cell scrolls onto the screen which might be a bit much, is there a better place to put the loading?

Reply:
this is loading the image file in the background and will display it when complete 

from the quick help: 
Once the download completes, the remote image will be displayed.